### PR TITLE
[TACHYON-626] Add a check for NULL BlockMetadataManagerView

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-626

### CODE REVIEW

**Comment by gpang** [comment]
This means the allocators and evictors don't need to do the check again, right? If so, the check should be removed from the individual allocators and evictors.

**Comment by miasantreble** [comment]
fine, my thought was that it won't hurt to check again but I will remove the check

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 1]
```diff
@@ -85,7 +85,8 @@
 
   public TieredBlockStore(TachyonConf tachyonConf) throws IOException {
     mTachyonConf = Preconditions.checkNotNull(tachyonConf);
-    mMetaManager = BlockMetadataManager.newBlockMetadataManager(mTachyonConf);
+    mMetaManager =
+        Preconditions.checkNotNull(BlockMetadataManager.newBlockMetadataManager(mTachyonConf));
```
Comment (uronce-cc) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: actually, I don't think the return value of `newBlockMetadataManager` here will be NULL anyway. What about checking the NULL metaDataManager in `AllocatorFactory` and `EvictorFactory` ?

Comment (miasantreble) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: make sense, you mean move the check to here, right? 
https://github.com/amplab/tachyon/blob/master/servers/src/main/java/tachyon/worker/block/allocator/AllocatorFactory.java#L32
https://github.com/amplab/tachyon/blob/master/servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java#L33

Comment (uronce-cc) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: yeah.

**Comment by haoyuan** [comment]
We can’t automatically merge this pull request.

**Comment by miasantreble** [comment]
rebased from master, merge conflict resolved

File: servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java [hunk 2]
```diff
@@ -30,7 +32,8 @@ private EvictorFactory() {}
    * @param view BlockMetadataManagerView to pass to Evictor
    * @return the generated Evictor
    */
-  public static Evictor create(EvictorType evictorType, BlockMetadataManagerView view) {
+  public static Evictor create(EvictorType evictorType, BlockMetadataManagerView bmmView) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java]: Could you change the argument name to `view` to be consistent with the rest of the codebase?

**Comment by calvinjia** [comment]
Looks good overall, minor style comment.

File: servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java (hunk 2 above)
Comment (miasantreble) [inline servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java]: done

**Comment by calvinjia** [comment]
+1 LGTM

**Comment by apc999** [comment]
+1 LGTM. though this PR will conflict with a few pending PRs --- so depending on the merge order, we need to resolve conflicts carefully to avoid regression.

**Comment by miasantreble** [comment]
thanks for speedy approval!

**Comment by haoyuan** [comment]
Thanks @miasantreble  :)

**Pull request merged**